### PR TITLE
Added express routes for cart, line items and categories

### PR DESCRIPTION
**Pull request by ami-domadia**

### CODE REVIEW

**Review by 5t4mp5 — COMMENTED** [review]
Do we have a way to get all LineItems for a particular cart? A line item get route to /:cartId might be useful for populating a user's cart.

File: server/routes/category.js [hunk 1]
```diff
@@ -4,6 +4,34 @@ const Category = db.Category;
...
 const router = express.Router();
 
+router.get('/', (req, res, next) => {
+    Category.findAll()
+    .then((categories) => res.json(categories))
+    .catch(next);
+});
 
+router.get('/:id', (req, res, next) => {
+    if(req.params.id){
```
Comment (chem2016) [inline server/routes/category.js]: do we need to do this logic here?

File: server/routes/category.js [hunk 2]
```diff
@@ -4,6 +4,34 @@ const Category = db.Category;
...
+});
+
+router.post('/', (req, res, next) => {
+    Category.create(req.body)
+    .then((category) => res.json(category))
+    .catch(next);
+});
+
+router.delete('/:id', (req, res, next) => {
+    if(req.params.id){
```
Comment (chem2016) [inline server/routes/category.js]: Same here, probably we do not need to do if statement here.

File: server/routes/lineitem.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const router = express.Router();
+
+router.get('/', (req, res, next) => {
+    LineItem.findAll()
+    .then((lineitems) => res.json(lineitems))
+    .catch(next);
+});
+
+router.get('/:id', (req, res, next) => {
+    if(req.params.id){
```
Comment (chem2016) [inline server/routes/lineitem.js]: do not need to do if statement here

File: server/routes/lineitem.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+router.post('/', (req, res, next) => {
+    LineItem.create(req.body)
+    .then((lineitem) => res.json(lineitem))
+    .catch(next);
+});
+
+router.delete('/:id', (req, res, next) => {
+    if(req.params.id){
```
Comment (chem2016) [inline server/routes/lineitem.js]: do not need to do if statement here

File: server/routes/category.js (hunk 1 above)
Comment (ami-domadia) [reply server/routes/category.js]: No, I removed it now.  Can look into error handling later.

File: server/routes/lineitem.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+router.get('/:id', (req, res, next) => {
+    LineItem.findOne({where: {id: req.params.id}})
+        .then((lineitem) => res.json(lineitem))
+        .catch(next);
+    
+});
+
+router.get('/:cartid', (req, res, next) => {
+    LineItem.findAll({where: {cartId: req.params.id}})
```
Comment (5t4mp5) [inline server/routes/lineitem.js]: req.params.id should be req.params.cartid, as you have the route written. We should probably also camel case that as cartId.

**Review by 5t4mp5 — CHANGES_REQUESTED** [review]

File: server/routes/lineitem.js (hunk 5 above)
Comment (ami-domadia) [reply server/routes/lineitem.js]: Ah thanks for catching that!

**Review by ghost — DISMISSED** [review]
Please merge.

**Review by 5t4mp5 — DISMISSED** [review]

**Review by chem2016 — COMMENTED** [review]
@ami-domadia, could you delete some of the empty lines in the routes. I will test the routes and then approve it.

**Review by chem2016 — COMMENTED** [review]
@ami-domadia, when I try to do get route /auth/session, it shows an 404 route not found? Is this the expected behavior?

**Comment by ami-domadia** [comment]
Hi @chem2016 Yes I believe it is hard to test GET /auth/session since the req.session has to be set.  And mocking that is not trivial I think.  I was also missing npm install express-session and morgan.  Just added that.  

@5t4mp5  Resolved conflicts for dbsync().
I made an update to the seed file to make testing easier for email and password.

**Review by 5t4mp5 — APPROVED** [review]

**Review by chem2016 — APPROVED** [review]
Thanks for making changes!

**Pull request merged**